<launch>


    <param name="model_path" value="$(find ymbot_amp_devel)/robots/unitree_g1/mjcf/g1_23dof_personal.xml" />
    <param name="onnx_path" value="$(find ymbot_amp_devel)/robots/unitree_g1/onnx/policy.onnx" />

    <!-- isaaclab 关节参数 -->
    <param name="num_joints" value="23" />

    <param name="joint_names" value='["left_hip_pitch_joint", "right_hip_pitch_joint", "waist_yaw_joint",
                                     "left_hip_roll_joint", "right_hip_roll_joint", "waist_roll_joint",
                                     "left_hip_yaw_joint", "right_hip_yaw_joint", "waist_pitch_joint",
                                     "left_knee_joint", "right_knee_joint", "left_shoulder_pitch_joint",
                                     "right_shoulder_pitch_joint", "left_ankle_pitch_joint", "right_ankle_pitch_joint",
                                     "left_shoulder_roll_joint", "right_shoulder_roll_joint", "left_ankle_roll_joint",
                                     "right_ankle_roll_joint", "left_shoulder_yaw_joint", "right_shoulder_yaw_joint",
                                     "left_elbow_joint", "right_elbow_joint"]'/>

    <param name="kps" value="[200, 200, 150, 150, 150, 150, 150, 150, 150, 200, 
                             200, 40, 40, 20, 20, 40, 40, 20, 20, 40, 40, 40, 40]"/>
    <param name="kds" value="[5, 5, 6, 5, 5, 6, 5, 5, 6, 5, 
                             5, 10, 10, 2, 2, 10, 10, 2, 2, 10, 10, 10, 10]"/>
    <param name="offsets" value="[-0.1, -0.1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 
                                0.3, 0.0, 0.0, -0.2, -0.2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.87, 0.87]"/>
    <param name="tau_limits" value="[200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 
                                   200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200]"/>
    <param name="directions" value="[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
                                   1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]"/>

    <!-- 定义 mujoco 关节名称 -->
    <param name="joint_names_mujoco" value='["left_hip_pitch_joint", "left_hip_roll_joint", "left_hip_yaw_joint",
                                            "left_knee_joint", "left_ankle_pitch_joint", "left_ankle_roll_joint",
                                            "right_hip_pitch_joint", "right_hip_roll_joint", "right_hip_yaw_joint",
                                            "right_knee_joint", "right_ankle_pitch_joint", "right_ankle_roll_joint",
                                            "waist_yaw_joint", "waist_roll_joint", "waist_pitch_joint",
                                            "left_shoulder_pitch_joint", "left_shoulder_roll_joint", "left_shoulder_yaw_joint",
                                            "left_elbow_joint", "right_shoulder_pitch_joint", "right_shoulder_roll_joint",
                                            "right_shoulder_yaw_joint", "right_elbow_joint"]'/>



    <node pkg="ymbot_amp_devel" type="amp_sim2sim" name="amp_sim2sim" output="screen" />

</launch>

    